ehsm_kms_service/quote_manager_apis.js

```diff
@@ -0,0 +1,256 @@
+ *          ==> {r}mrsigner(String [1~1024]) : a base64 string. eg. 'NzAwMDAwMDAyMzEwMDAwMDAwMTM1M****'
+ * @returns {Object}
+ *          ==> {r}policyId(String) : a new policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ */
+const uploadQuotePolicy = async (res, appid, payload, DB) => {
+    try {
+        const { mrenclave, mrsigner } = payload
+        if (!checkStringParam(mrenclave, true, QUOTE_LENGTH_MAX)) {
+            res.send(_result(400, `mrenclave cannot be empty, must be string and length not more than ${QUOTE_LENGTH_MAX}`))
+            return
+        }
+        if (!checkStringParam(mrsigner, true, NONCE_LENGTH_MAX)) {
+            res.send(_result(400, `mrsigner cannot be empty, must be string and length not more than ${NONCE_LENGTH_MAX}`))
+            return
+        }
+        let policyId = uuidv4();
+        await DB.insert({
+            _id: `quote_policy:${policyId}`,
+            appid,
+            policyId,
+            mrenclave,
+            mrsigner
+        })
+        let result = {
+            policyId
+        }
+        res.send(_result(200, 'Upload quote policy success.', result))
+        return
+    } catch (e) {
+        console.info('uploadQuotePolicy :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * query a quote policy information by policyid.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}policyId(String) : a policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ * @returns {Object}
+ *          ==> {r}policyId(String) : a policy ID. eg. '15ca0dd5-2d34-4221-a708-3171ffe6****'
+ *          ==> {r}mrenclave(String) : a base64 string. eg. 'NTU4OTI1NTI1NTAwMDAwMDAwMDAw****'
+ *          ==> {r}mrsigner(String) : a base64 string. eg. 'NzAwMDAwMDAyMzEwMDAwMDAwMTM1M****'
+ */
+const getQuotePolicy = async (res, appid, payload, DB) => {
+    try {
+        const { policyId } = payload
+        if (!checkStringParam(policyId, true, 36)) {
+            res.send(_result(400, `policyId cannot be empty, must be string and length not more than 36.`))
+            return
+        }
+        if (!check_UUID_format(policyId)) {
+            res.send(_result(400, 'policyId format wrong.'))
+            return
+        }
+
+        let result = {
+            policyId
+        }
+        const query = {
+            selector: {
+                appid,
+                policyId
+            },
+            fields: ['mrenclave', 'mrsigner'],
+            limit: 1
+        }
+        let quote_policy_res = await DB.partitionedFind('quote_policy', query)
+        if (quote_policy_res.docs.length > 0) {
+            result['mrenclave'] = quote_policy_res.docs[0].mrenclave
+            result['mrsigner'] = quote_policy_res.docs[0].mrsigner
+        }
```

Comment:
the result may also need to inlcude the create time

Reply:
ok,will add createTime